mreg/api/permissions.py

```diff
@@ -10,7 +56,142 @@ class IsInRequiredGroup(BasePermission):
     def has_permission(self, request, view):
         if not bool(request.user and request.user.is_authenticated):
             return False
-        REQUIRED_USER_GROUP = getattr(settings, 'REQUIRED_USER_GROUP', None)
-        if REQUIRED_USER_GROUP is None:
-            raise exceptions.APIException(detail='REQUIRED_USER_GROUP is unset')
-        return request.user.groups.filter(name=REQUIRED_USER_GROUP).exists()
+        return user_in_settings_group(request, 'REQUIRED_USER_GROUPS')
+
+
+class ReadOnlyForRequiredGroup(IsInRequiredGroup):
+    """
+    Allows read only access to users in the required group.
+    """
+
+    def has_permission(self, request, view):
+        if super().has_permission(request, view):
+            return request.method in SAFE_METHODS
+        return False
+
+
+class IsSuperGroupMember(BasePermission):
+    """
+    Permit user if in super user group.
+    """
+
+    group = 'SUPERUSER_GROUP'
+
+    def has_permission(self, request, view):
+        if not bool(request.user and request.user.is_authenticated):
+            return False
+        return user_in_settings_group(request, 'SUPERUSER_GROUP')
+
+
+class IsSuperOrAdminOrReadOnly(BasePermission):
+    """
+    Permit user if in super or admin group, else read only.
+    """
+
+    def has_permission(self, request, view):
+        if not bool(request.user and request.user.is_authenticated):
+            return False
+        if request.method in SAFE_METHODS:
+            return True
+        self.user_groups = set(request.user.groups.values_list('name', flat=True))
```

Comment:
Considering moving this to a convenience property on the user object

Reply:
Maybe a new issue? Would cause a lot more churn..